java/arcs/core/storage/StorageKey.kt

```diff
@@ -24,4 +24,12 @@ abstract class StorageKey(val protocol: String) {
         childKeyWithComponent("handle/$handleId")
 
     override fun toString(): String = "$protocol://${toKeyString()}"
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
```

Comment:
This won't compile cross-platform..   I suggest just doing `if (other == null || other::class != this::class) return false`

Reply:
Done